src/serializer/visitors.js

```diff
@@ -94,6 +94,48 @@ export let ClosureRefReplacer = {
       }
     }
   },
+
+  // A few very simple dead code elimination helpers. Eventually these should be subsumed by the partial evaluators.
+  IfStatement: {
+    exit: function(path: BabelTraversePath, state: ClosureRefReplacerState) {
+      let node = path.node;
+      if (t.isBooleanLiteral(node.test)) {
```

Comment:
You could generalize all cases a bit for other kinds of literals that have truthy values.

Reply:
Sure, done. This is getting further and further into the realm of partial evaluation, though - for example, I'm considering `[]` to be truthy, but not `[0]` (because I don't want to eliminate `[sideEffect()]`, and don't have a 'isPossiblySideEffecting' function on hand).